src/contracts/Staking.sol

```diff
@@ -113,13 +155,92 @@ contract Staking is Ownable {
         });
     }
 
+    /**
+        @notice claim TOKE from Tokemak
+        @param _claimAddress address
+        @param _amount uint
+        @param _v uint
+        @param _r bytes
+        @param _s bytes
+     */
+    function claimFromTokemak(
+        address _claimAddress,
+        uint256 _amount,
+        uint8 _v,
+        bytes32 _r,
+        bytes32 _s
+    ) public onlyManager {
+        ITokeReward tokeRewardContract = ITokeReward(tokeReward);
+        ITokeRewardHash iTokeRewardHash = ITokeRewardHash(tokeRewardHash);
+
+        uint256 latestCycleIndex = iTokeRewardHash.latestCycleIndex();
+        Recipient memory recipient = Recipient({
+            chainId: 1,
+            cycle: latestCycleIndex,
+            wallet: address(this),
+            amount: _amount
+        });
+        tokeRewardContract.claim(recipient, _v, _r, _s);
+        IERC20(stakingToken).safeTransfer(_claimAddress, _amount);
+    }
+
+    /**
+        @notice get claimable amount of TOKE from Tokemak
+        @param _amount uint
+        @return uint
+     */
+    function getClaimableAmountTokemak(uint256 _amount)
```

Comment:
can this be `external`, it will save gas if so.

Reply:
It can be, but this could also be moved to sdk